server/sonar-server/src/test/java/org/sonar/server/rule/RegisterRulesTest.java

```diff
@@ -661,6 +661,35 @@ public void declaring_a_rule_with_a_deprecated_RuleKey_still_used_should_throw_I
     });
   }
 
+  @Test
+  public void declaring_a_rule_with_an_existing_RuleKey_still_used_should_throw_IAE() {
```

Comment:
I'm not seeing the test case with the extended repository

you mentionned it and I agree it should be nice to have it